lib/svg2png.js

```diff
@@ -1,75 +1,53 @@
 "use strict";
-const path = require("path");
+
 const fileURL = require("file-url");
-const childProcess = require("pn/child_process");
+const puppeteer = require("puppeteer");
 
-const phantomjsCmd = require("phantomjs-prebuilt").path;
-const converterFileName = path.resolve(__dirname, "./converter.js");
+const converter = require("./converter");
 
 const PREFIX = "data:image/png;base64,";
 
 module.exports = (sourceBuffer, options) => {
     return Promise.resolve().then(() => { // catch thrown errors
-        const cp = childProcess.execFile(phantomjsCmd, getPhantomJSArgs(options), { maxBuffer: Infinity });
-
-        writeBufferInChunks(cp.stdin, sourceBuffer);
+        options = parseOptions(options);
 
-        return cp.promise.then(processResult);
+        return puppeteer.launch()
```

Comment:
Let's just use `async/await`. Also, please never use `Promise.all` in this fashion (i.e. with one thing that's a promise and one thing that's not, just so that you can pass them around).

I also see no reason why `convert` returns the browser that it was given. We can just keep using the same one throughout.

Reply:
I was very tempted to use `async/await` but I held myself back since your README explicitly stated support for Node.js 6, which wouldn't be possible if we adopted `async/await`. If you're happy to drop support for Node.js 6, I can rewrite using `async/await` and it will be much more readable.

Without using `async/await` I was trying to avoid holding references/state outside of the promise flow as I felt that was an anti-pattern. If we're going to use `async/await`, there will be no need for these. Otherwise, I'll rewrite to store references at the function level to avoid passing references.

Reply:
Yep, let's drop Node 6; this will be a major version bump anyway.